pkg/apis/integreatly/v1alpha1/rhmi_types.go

```diff
@@ -57,6 +58,7 @@ var (
 	// It is preferable to determine the version of a product dynamically (from a CR, or configmap, etc)
 	// Follow up Jira: https://issues.redhat.com/browse/INTLY-5946
 	VersionAMQOnline           ProductVersion = "1.4"
+	VersionApicurioRegistry    ProductVersion = "1.2.3.final"
```

Comment:
yes, agreed

Reply:
Done.

Reply:
I think this change might have broken the code.

For what I see, as it is done right now VersionApicurioRegistry is used in ApicurioRegistry reconciler code as the container image tag of the Apicurio Registry streams image. There's no "1.2.3" tag so the image will fail to be pulled by Apicurio operator.

Additionally, after reviewing this part of code, I'm not sure specifying the image in the Apicurio CR is a good idea, and I'm also not convinced on pulling a docker.io image. I think a productised one should be used.

If I remember correctly apicurio registry operator has code setting a default apicurio registry image, and in case of productized versions of Apicurio registry operator, the image url value of the apicurio operand is set in the CSV if I remember correctly so we should not need to specify it in the CR. In fact, I'd say it's not even desirable that we do set it in integreatly code.
If the CSV of apicurio registry that we are including here as a copy is a productized one (the one that is used in olm catalog) the envvar in the CSV will be set to the appropriate operand image productized value.

cc @akieling, @philbrookes, @davidffrench 

If we decide to go this route it will also mean that an image from registry.redhat.io image will be pulled by the Apicurio registry operator.

Reply:
Thanks for the review, @miguelsorianod . Yes, currently the `VersionApicurioRegistry` variable in `rhmi_types.go` is used to create the product image URL in the CR. That CR field might not be required anymore. I'll confirm that with Jakub Senko.

Reply:
I'll undo my change and keep the real version name. It's also used in the `VerifyVersion` function in the reconciler. The reference to the product image name can be removed from the ApicurioRegistry CR as confirmed with Jakub.

Reply:
Is it OK to keep the version `1.2.3.final`?

Reply:
If it needs to be this way, this is OK.